### DISCUSSION TITLE
Slider with long labels

### DISCUSSION THREAD

**Discussion by smartm0use** [fomantic/Fomantic-UI#1355] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hello, I need to create a slider that has steps with long labels.

Having a look at this https://jsfiddle.net/h98vpkfs I mean replacing 5, 10, 15 etc. with something like "First day of something" or "Day of some event" and so on.

Is there any way to reach that? Maybe with tooltip, but how?

**Comment by jamessampford** [comment]
This sounds better suited for a range calendar as between 2 dates?

https://fomantic-ui.com/modules/calendar.html#range

**Comment by smartm0use** [comment]
Yes, but I need to have it as a slider, as used in websites like skyscanner.
(Anyway, even with calendar I do not have ways to show long labels instead of the number of the day).

**Comment by lubber-de** [comment]
You can use the custom interpreted labels feature https://fomantic-ui.com/modules/slider.html#custom-interpreted-labels
See http://jsfiddle.net/L4htmsno/
```javascript
var labels = [
	"First day of something",
  "In the middle of nowhere",
  "Lost in space",
  "End of the world"
];
$('#interpretedlabel')
  .slider({
   	min: 0,
    max: 3,
    start: 0,
    end: 3,
    step: 1,
    interpretLabel: function(value) {
      return labels[value];
    }
  })
;
```
![longsliderlabels](https://user-images.githubusercontent.com/18379884/76101233-632cbf00-5fce-11ea-9e62-9eb51b0fb8d2.gif)

**Comment by smartm0use** [comment]
Thank you for your suggestion, but I'm afraid I can't use this because I have to create a very small slider:

![download](https://user-images.githubusercontent.com/5399265/76101650-0e3d7880-5fcf-11ea-99bc-0fa76e769872.png)

Can't I use tooltip in this case?

**Comment by lubber-de** [comment, marked as answer]
Sure 😄 
See http://jsfiddle.net/nos06d5e/
![longsliderlabelstooltips](https://user-images.githubusercontent.com/18379884/76102679-a12ae280-5fd0-11ea-9760-d8db74baf9d0.gif)

**Comment by smartm0use** [comment]
Wow, perfect! Thank you so much!

**Comment by smartm0use** [comment]
Shouldn't I add an input tag like this:

```
<div class="ui input">
  <input type="text" id="slider-input-1" disabled="">
</div>
```

As seen [here](https://fomantic-ui.com/modules/slider.html#slider)?

**Comment by lubber-de** [comment]
The input tag is only there to show the values in the docs. To fetch the values you use the `onChange` callback or the `get value` behavior

**Comment by smartm0use** [comment]
In a range slider how to get from the `onChange` callback the range values, instead of value of the last move? I mean the min and max value selected by the user.

In addition, since I'm wrapping this in an Angular component, do you think I can add an hidden input element to bind it to a formControl?

**Comment by smartm0use** [comment]
I have edited my last post, sorry for double posting now.

**Comment by lubber-de** [comment]
> In a range slider how to get from the `onChange` callback the range values, instead of value of the last move? I mean the min and max value selected by the user.

The `onChange` callback provides 3 parameters: range (= difference between firstValue and lastValue), firstValue, lastValue

```javascript
onChange: function(range, firstVal, secondVal) {
}
```

**Comment by smartm0use** [comment]
Great! Thank you, I cannot find it in the documentation.

**Comment by smartm0use** [comment]
Hello again, sorry for keeping post here but I have one more question.
I noticed that the `onChange` callback is triggered even if the slider is not moved.
Both `onChange` and `onMove` are triggered just clicking on the slider, without any values change.
How to trigger onChange only when the value are different from previous change?

**Comment by lubber-de** [comment]
Sounds like a bug we have to investigate. Please create a new issue for that. Thanks
